### PR TITLE
update rust_analyzer_settings.json

### PR DESCRIPTION
**Pull request by RalfJung**
This works around https://github.com/rust-lang/rust-analyzer/issues/15595, and avoids relying on the unspecified working directory of this command.

### CODE REVIEW

File: src/etc/rust_analyzer_settings.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
         "compiler/rustc_codegen_gcc/Cargo.toml"
     ],
     "rust-analyzer.rustfmt.overrideCommand": [
-        "./build/host/rustfmt/bin/rustfmt",
+        "${workspaceFolder}/build/host/rustfmt/bin/rustfmt",
         "--edition=2021"
     ],
     "rust-analyzer.procMacro.server": "./build/host/stage0/libexec/rust-analyzer-proc-macro-srv",
```
Comment (RalfJung) [inline src/etc/rust_analyzer_settings.json:22]: @Veykril  will that variable also work for this setting? What about `check.overrideCommand` and `buildScripts.overrideCommand`? We should probably use it everywhere?

Comment (Veykril) [reply src/etc/rust_analyzer_settings.json:22]: Yes it should work in all VSCode settings.

**Comment by lqd** [comment]
There was also https://github.com/rust-lang/rust/pull/115818

**Comment by RalfJung** [comment]
Ah, looks like we raced. I took the opportunity to also change `procMacro.server`, but I'm fine either way -- no strong preference for which PR to land.

**Pull request merged**